dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/UnitOfWorkAspect.java

```diff
@@ -80,7 +77,16 @@ public void onError() {
         try {
             rollbackTransaction();
         } finally {
-            session.close();
+            onFinish();
+        }
+    }
+
+    public void onFinish() {
```

Comment:
There is a possibility, but I think we can presume that Jersey will always generate a `FINISHED` event, and `onFinish` will be called and it will clean up resources.